helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTask.java

```diff
@@ -168,7 +169,14 @@ public HelixTaskResult call() {
 
       // forward relay messages attached to this message to other participants
       if (taskResult.isSuccess()) {
-        forwardRelayMessages(accessor, _message, taskResult.getCompleteTime());
+        try {
+          forwardRelayMessages(accessor, _message, taskResult.getCompleteTime());
+        } catch (Exception e) {
+          // Fail to send relay message should not result in a task execution failure
+          // Currently we don't log error to ZK to reduce writes as when accessor throws
+          // exception, ZK might not be in good condition.
+          logger.error("Failed to send relay messages.", e);
```

Comment:
logger.warn should be fine, this is not fatal error.

Reply:
will change